go/adbc/drivermgr/wrapper_duckdb_test.go

```diff
@@ -0,0 +1,163 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build cgo
+
+package drivermgr_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-adbc/go/adbc/drivermgr"
+	"github.com/apache/arrow/go/v16/arrow"
+	"github.com/apache/arrow/go/v16/arrow/array"
+	"github.com/stretchr/testify/suite"
+	"github.com/substrait-io/substrait-go/extensions"
+	"github.com/substrait-io/substrait-go/plan"
+	"github.com/substrait-io/substrait-go/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type DriverMgrDuckDBSuite struct {
+	suite.Suite
+
+	ctx  context.Context
+	drv  adbc.Driver
+	db   adbc.Database
+	conn adbc.Connection
+}
+
+func (dm *DriverMgrDuckDBSuite) SetupSuite() {
+	dm.ctx = context.TODO()
+	dm.drv = drivermgr.Driver{}
+	var err error
+	dm.db, err = dm.drv.NewDatabase(map[string]string{
+		"driver":     "libduckdb",
+		"entrypoint": "duckdb_adbc_init",
+	})
+	dm.NoError(err)
+
+	cnxn, err := dm.db.Open(dm.ctx)
+	dm.NoError(err)
+	defer cnxn.Close()
+
+	stmt, err := cnxn.NewStatement()
+	dm.NoError(err)
+	defer stmt.Close()
+
+	// Setup substrait extension
+	dm.NoError(stmt.SetSqlQuery("INSTALL substrait"))
+	_, err = stmt.ExecuteUpdate(dm.ctx)
+	dm.NoError(err)
+
+	dm.NoError(stmt.SetSqlQuery("LOAD substrait"))
+	_, err = stmt.ExecuteUpdate(dm.ctx)
+	dm.NoError(err)
+
+	dm.NoError(stmt.SetSqlQuery("CREATE TABLE test_table (id INTEGER PRIMARY KEY, name TEXT)"))
+	// Don't check nrows, duckdb currently does not support it
+	_, err = stmt.ExecuteUpdate(dm.ctx)
+	dm.NoError(err)
+
+	dm.NoError(stmt.SetSqlQuery("INSERT INTO test_table (id, name) VALUES (1, 'test')"))
```

Comment:
any reason to not just insert all three rows in one query?

`INSERT INTO test_table (id, name) VALUES (1, 'test'), (2, 'test'), (3, 'tested')` or does DuckDB not support that?